ext/mbstring/libmbfl/filters/mbfilter_utf7imap.c

```diff
@@ -732,3 +734,135 @@ static void mb_wchar_to_utf7imap(uint32_t *in, size_t len, mb_convert_buf *buf,
 
 	MB_CONVERT_BUF_STORE(buf, out, limit);
 }
+
+static bool is_base64_end_valid(unsigned char n, bool gap, bool is_surrogate)
+{
+	return !(gap || is_surrogate || n == ILLEGAL);
+}
+
+
+static bool is_utf16_cp_valid(uint16_t cp, bool is_surrogate)
+{
+	if (is_surrogate) {
+		return cp >= 0xDC00 && cp <= 0xDFFF;
+	} else if (cp >= 0xDC00 && cp <= 0xDFFF) {
+		/* 2nd part of surrogate pair came unexpectedly */
+		return false;
+	} else if (cp >= 0x20 && cp <= 0x7E && cp != '&') {
+		return false;
+	}
+	return true;
+}
+
+static bool has_surrogate(uint16_t cp, bool is_surrogate)
+{
+	return !is_surrogate && cp >= 0xD800 && cp <= 0xDBFF;
```

Comment:
These functions seem to be common with ``ext/mbstring/libmbfl/filters/mbfilter_utf7.c`` would it make sense to put them in a separate file and header include them?

Reply:
I would even suggest combining UTF-7 and UTF7-IMAP in a single `.c` file, though maybe that could be done as a follow-up PR. If these functions are moved into the header file, they should be `static inline`.

Reply:
Combining them is also a good idea.

Reply:
> I can't have an opinion on file combining because I don't understand why existing files are merged.

Originally, almost all the text encodings supported by libmbfl (and thus mbstring) were implemented in separate files. There were a couple exceptions, but basically it was one `.c` file per supported text encoding.

I have combined some of them to facilitate sharing code and thus reducing the total code size of the library. I would also like to combine more files in the future.

Don't know if this makes things clearer or not.